projects/storefrontlib/src/lib/checkout/guards/shipping-address-set.guard.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { CheckoutDetailsService } from '../checkout-details.service';
+import { CheckoutStep } from '../config/model/checkout-step.model';
+import { CheckoutConfig } from '../config/checkout-config';
+
+@Injectable()
+export class ShippingAddressSetGuard implements CanActivate {
+  constructor(
+    private checkoutDetailsService: CheckoutDetailsService,
+    private router: Router,
+    private config: CheckoutConfig
+  ) {}
+
+  canActivate(): Observable<boolean | UrlTree> {
+    const route = this.config.checkout.steps.find(
```

Comment:
Add console.warn for better DX

Reply:
should I use ServerConfig (from '../../config/server-config/server-config') to check production flag?

Reply:
yes